sdk/src/schema.ts

```diff
@@ -71,30 +70,16 @@ export type InvariantEventType =
   | RemovePositionEvent
   | SwapEvent
 
-export interface TxResult {
-  hash: string
-  events: InvariantEventType[]
-}
-
-export type CreatePositionTxResult = {
-  hash: string
-  events: [CreatePositionEvent]
-}
-
-export type RemovePositionTxResult = {
-  hash: string
-  events: [RemovePositionEvent]
-}
-
-export type SwapTxResult = {
+export interface TxResultInterface<T> {
   hash: string
-  events: [CrossTickEvent, SwapEvent] | [SwapEvent]
+  events: T
 }
 
-export type SwapRouteTxResult = {
-  hash: string
-  events: (CrossTickEvent | SwapEvent)[]
-}
+export type TxResult = TxResultInterface<InvariantEventType[]>
```

Comment:
There should also be a result type for responses without events. While it's possible to use an empty array, this follows a soft-typed style.

I suggest using the `extends` interface to add an `events` field - this solves the problem, and we do not compromise with generics

`TxResult {hash}`
`EventTxResult extends TxResult {events}`